### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by midasdev711**
### Problem

- In the ultrawide monitor, it's kind of hard to read all the numbers as they are really separated and all over the place.
- Top left corner logo is blurry.
- Nav bar buttons don't have action when hovering.
- It makes no sense when the mouse is on top of #2 row.

### Solution

- Set the max-width of the content
- Replace the logo
- Change color of nav bar link when hovering
- Remove hover effect in all tables

### Result

All issues are fixed.

image

### CODE REVIEW

File: web-ui/src/environments/environment.ts
Comment (adinael) [inline web-ui/src/environments/environment.ts:9]: this line should not be changed.

File: web-ui/angular.json
Comment (AlexITC) [inline web-ui/angular.json]: Mixing jquery with angular is usually a bad idea, what approaches have you considered that led you to still require jquery?

File: web-ui/angular.json
Comment (AlexITC) [inline web-ui/angular.json]: I think the explorer already uses https://valor-software.com/ngx-bootstrap/#/, what's the need behind bringing bootstrap separately?

File: web-ui/angular.json
Comment (AlexITC) [inline web-ui/angular.json:25]: Is this component ever used?

File: web-ui/package.json
Comment (AlexITC) [inline web-ui/package.json]: Add a new command instead

File: web-ui/package.json
Comment (AlexITC) [inline web-ui/package.json]: Rename this to `buiild-prod` instead

File: web-ui/package.json
Comment (AlexITC) [inline web-ui/package.json:17]: While this is nice, you will likely have a hard time getting the tests to pass, what has forced you to upgrade?

File: web-ui/package.json
Comment (AlexITC) [inline web-ui/package.json:28]: How is this being used?

File: web-ui/package.json
Comment (AlexITC) [inline web-ui/package.json:31]: How is this being used?

File: web-ui/package.json
Comment (AlexITC) [inline web-ui/package.json:33]: This is a backend dependency being added to a frontend app, I'm interested to know if this is really required and why

File: web-ui/package.json
Comment (AlexITC) [inline web-ui/package.json]: As pointed out in another commnet, jquery is usually not recommended in angular projects, what prevent us to remove it?

File: web-ui/package.json
Comment (AlexITC) [inline web-ui/package.json:34]: Can you give me some examples where this library is used?

File: web-ui/server.js
Comment (AlexITC) [inline web-ui/server.js:5]: Isn't this already handled by angular?

File: web-ui/tsconfig.json
Comment (AlexITC) [inline web-ui/tsconfig.json:10]: From the official [docs](https://www.typescriptlang.org/tsconfig#skipLibCheck), its preferred to make sure that the same library has a single version in our code, try that instead of adding this config

File: web-ui/src/styles.css
Comment (AlexITC) [inline web-ui/src/styles.css:309]: Wouldn't it be better to take this svg to its own file?

File: web-ui/src/styles.css
Comment (AlexITC) [inline web-ui/src/styles.css:239]: `!important` is something that is generally avoided unless there is no other way, I see quite a lot of times this is being used, please consider avoiding it

File: web-ui/src/index.html
Comment (AlexITC) [inline web-ui/src/index.html:32]: Try not mixing two ways for importing scripts, npm is the way we are using, also, npm imports jquery which is imported again.

File: web-ui/src/assets/Updating.json
Comment (AlexITC) [inline web-ui/src/assets/Updating.json:1]: What's the purpose of this file? also, make sure to be consistent with the names, this is the only one starting with upper case.

File: web-ui/src/app/app.module.ts
Comment (AlexITC) [inline web-ui/src/app/app.module.ts]: Remove if this is not used

File: web-ui/src/app/app.component.html
Comment (AlexITC) [inline web-ui/src/app/app.component.html:8]: I think that `http-loader` doesn't work anymore

File: web-ui/src/app/utils.ts
Comment (AlexITC) [inline web-ui/src/app/utils.ts:8]: Typescript is useless if you use everything as `any`, make sure to specify types.

File: web-ui/src/app/utils.ts
Comment (AlexITC) [inline web-ui/src/app/utils.ts:19]: Use a function name that's clear enough, also, include tests.

If this function was copied from somewhere else, add the source in a comment, also, consider using moment.js which already handles the corner cases properly

File: web-ui/src/app/utils.ts
Comment (AlexITC) [inline web-ui/src/app/utils.ts:43]: Typescript is useless if you use everything as `any`, make sure to specify types.

Also, add tests

File: web-ui/src/app/utils.spec.ts
Comment (AlexITC) [inline web-ui/src/app/utils.spec.ts:4]: Add better tests

File: web-ui/src/app/pipes/explorer-amount.pipe.ts
Comment (AlexITC) [inline web-ui/src/app/pipes/explorer-amount.pipe.ts]: This is not consistent to how the if-else are being written in the code, we need a code formatter

File: web-ui/src/app/pipes/explorer-amount.pipe.ts
Comment (AlexITC) [inline web-ui/src/app/pipes/explorer-amount.pipe.ts]: > fillZero:Boolean

Add space after the colon

File: web-ui/src/app/pipes/explorer-amount.pipe.ts
Comment (AlexITC) [inline web-ui/src/app/pipes/explorer-amount.pipe.ts]: Its not clear what this does, ideally, add a couple of tests

File: web-ui/src/app/pipes/pipes.module.ts
Comment (AlexITC) [inline web-ui/src/app/pipes/pipes.module.ts:13]: I don't see how the pipes module could depend on forms/router/etc, make sure to do some clean up in the other modules too

**Review by AlexITC — CHANGES_REQUESTED** [review]
I left some preliminary comments, overall, please take the comments for anything else that I haven't reviewed, these are some patterns I found:
- Its not clear that every dependency is necessary, from the npm ones to the modules depending on other modules.
- The code style is not consistent to what is already in the code, sometimes it isn't even consistent with the new code being written.
- Some function/variable names aren't clear.
- Lack of testing for important functions like money transformation, while I don't expect extensive testing, at least the core functionality can have a decent coverage.
- Usage of `any` when it doesn't seem necessary, Typescript is useless in this case.


Thanks.

File: web-ui/src/app/pipes/explorer-amount.pipe.ts
Comment (midasdev711) [reply web-ui/src/app/pipes/explorer-amount.pipe.ts]: That's for this case.

{{ 8456.789 | explorerAmount }}  => 8,456.789
{{ 8456.789 | explorerAmount:4:true }}  => 8,456.7890

File: web-ui/angular.json
Comment (midasdev711) [reply web-ui/angular.json:25]: Yes, in dex host calculator

File: web-ui/angular.json
Comment (midasdev711) [reply web-ui/angular.json]: I removed them.

File: web-ui/angular.json
Comment (midasdev711) [reply web-ui/angular.json]: Removed it.

File: web-ui/package.json
Comment (AlexITC) [inline web-ui/package.json:8]: Please explain what's the usefulness to include express because angular already has an embedded server for dev purposes, if this is not necessary, remove it

File: web-ui/src/app/components/addresses/address-details/address-details.component.ts
Comment (AlexITC) [inline web-ui/src/app/components/addresses/address-details/address-details.component.ts:28]: Use a enum instead of int values, or at least, create constants representing the possible statuses instead of using magic numbers

File: web-ui/src/app/components/addresses/address-details/address-details.component.ts
Comment (AlexITC) [inline web-ui/src/app/components/addresses/address-details/address-details.component.ts]: What's this for? you forgot to define the type

File: web-ui/src/app/components/addresses/address-details/address-details.component.ts
Comment (AlexITC) [inline web-ui/src/app/components/addresses/address-details/address-details.component.ts:27]: you likely need a `isLoading` per component loaded asynchronously, otherwise, there will be conflicts while loading transactions and tpos contracts

File: web-ui/src/app/components/addresses/address-details/address-details.component.ts
Comment (AlexITC) [inline web-ui/src/app/components/addresses/address-details/address-details.component.ts]: Remove the unused code instead of commenting it, unless there is a reason to keep, in which case, add the necessary comment

File: web-ui/src/app/components/blocks/block-list/block-list.component.ts
Comment (AlexITC) [inline web-ui/src/app/components/blocks/block-list/block-list.component.ts:22]: Same comment about magic numbers

File: web-ui/src/app/components/calculator/calculator.component.ts
Comment (AlexITC) [inline web-ui/src/app/components/calculator/calculator.component.ts]: Missing type

File: web-ui/src/app/components/calculator/calculator.component.ts
Comment (AlexITC) [inline web-ui/src/app/components/calculator/calculator.component.ts]: Make sure to address the missing type specified in most of the changes

File: web-ui/src/app/components/calculator/calculator.component.ts
Comment (AlexITC) [inline web-ui/src/app/components/calculator/calculator.component.ts:13]: This component has quite a lot of functions that could be taken out of the component, make sure to tests those, its pretty important to get the correct data for users

File: .circleci/config.yml
Comment (AlexITC) [inline .circleci/config.yml:21]: Tests are expected to run in the CI

File: web-ui/src/app/components/nodes/masternode-details/masternode-details.component.html
Comment (AlexITC) [inline web-ui/src/app/components/nodes/masternode-details/masternode-details.component.html:4]: Please make sure that all the changes have properly formatted code

**Review by AlexITC — CHANGES_REQUESTED** [review]
Overall I took a quick look to this huge change, I'd like to merge it as soon as possible, I'll approve when the PR gets these changes:
- Enable the tests and get them passing in the CI
- Format the code properly, there are some html pieces not aligned.
- Remove unused/commented code
- There are some critical pieces that I want you tor write tests for, like the calculator functions
- Try to avoid using `any` types, and specify types when possible.

Thanks.

File: web-ui/package.json
Comment (midasdev711) [reply web-ui/package.json:8]: That's for staging server. (http://staging.xsnexplorer.io/)
In the staging server, I build the project and run it with node server.
You can check that in server.js.

File: web-ui/src/app/components/calculator/calculator.component.ts
Comment (midasdev711) [reply web-ui/src/app/components/calculator/calculator.component.ts]: Added type.

File: web-ui/src/app/components/calculator/calculator.component.ts
Comment (midasdev711) [reply web-ui/src/app/components/calculator/calculator.component.ts]: Added type.

File: web-ui/src/app/components/addresses/address-details/address-details.component.ts
Comment (midasdev711) [reply web-ui/src/app/components/addresses/address-details/address-details.component.ts]: Added type.
That's to save interval id so that the component can remove interval when it's destroyed.

File: web-ui/package.json
Comment (AlexITC) [reply web-ui/package.json:8]: In my opinion, this is not necessary for the project, I'd suggest keeping this change in your local machine only

Comment (AlexITC) [reply web-ui/package.json:8]: In fact, given that it is staging, why can't you use `ng server` instead?

File: .circleci/config.yml
Comment (midasdev711) [reply .circleci/config.yml:21]: There is no e2e test in the project now. So I have to start from scratch.

Comment (midasdev711) [reply .circleci/config.yml:21]: Can you give me the scope of the test?

Comment (midasdev711) [reply .circleci/config.yml:21]: Which workflow should be tested with protractor?

Comment (AlexITC) [reply .circleci/config.yml:21]: Apologize, you are right that we don't have e2e tests, this is fine.

**Comment by AlexITC** [comment]
Superseded by #329 .

**Pull request closed without merging**
